tests/integration/utils.go

```diff
@@ -1116,9 +1116,12 @@ func patchCollection(
 	s *state.State,
 	action PatchCollection,
 ) {
-	_, nodes := getNodesWithIDs(action.NodeID, s.Nodes)
-	for _, node := range nodes {
+	nodeIDs, nodes := getNodesWithIDs(action.NodeID, s.Nodes)
+	for index, node := range nodes {
+		nodeID := nodeIDs[index]
+		s.Ctx = getContextWithIdentity(s.Ctx, s, action.Identity, nodeID)
```

Comment:
todo: I can't spot why you are overwriting `s.Ctx` here, and arent just doing the below instead of this set/rest pattern:
```
ctx := getContextWithIdentity(s.Ctx, s, action.Identity, nodeID)
err := node.PatchCollection(ctx, action.Patch, action.Lens)
```
I assume there are a bunch of places that could be simplified like this.